float64 t0
float64 tf
int32 poly_order
float64[] x_coeff
float64[] y_coeff
float64[] z_coeff